### DISCUSSION TITLE
Why the upper and lower triangles of a correlation matrix are different

### DISCUSSION THREAD

**Discussion by Owenxz** [AdaptiveMotorControlLab/CEBRA#155] [Q&A] [answered]
Hi CEBRA team! First of all, thanks for creating such a powerful and user-friendly tool. I'm using the consistency scores to see how stable the Cebra-behavior models trained on different animals are. I have seen some very different scores in the same pair of animals. I noticed the different scores in the demo as well, but they were less dramatic than mine. I'm assuming the difference came from the two labels for each animal, but the [documentation](https://cebra.ai/docs/api/sklearn/helpers.html) suggests that "all sets of digitalized labels contain the same set of values". I would appreciate if more information can be shared on how the consistency scores are calculated. Thanks!
![c6ce5b9cf8d3e48cf155ece0f49b5d50](https://github.com/AdaptiveMotorControlLab/CEBRA/assets/87492412/e14d0923-cad4-419c-8f32-7936bc8b73fd)

**Comment by stes** [comment, marked as answer]
Hi @Owenxz , for a list of embeddings X1, ..., Xn each entry in the matrix is the value of regressing from Xi -> Xj, and reporting the R2 value of that linear regression.

This R2 can be different for the Xi->Xj and Xj->Xi regression problems. If you are not interested in considering this difference, you could also compute the average between the upper and lower parts.

**Reply by Owenxz** [reply]
Thank you so much! That makes a lot of sense.
